eng/packages/http-client-csharp/generator/TestProjects/Local/Basic-TypeSpec/src/Generated/BasicTypeSpecClient.cs

```diff
@@ -127,12 +127,12 @@ public virtual Response<Thing> SayHi(string headParameter, string queryParameter
         /// <param name="optionalQuery"></param>
         /// <exception cref="ArgumentNullException"> <paramref name="headParameter"/> or <paramref name="queryParameter"/> is null. </exception>
         /// <exception cref="RequestFailedException"> Service returned a non-success status code. </exception>
-        public virtual async Task<Response<Thing>> SayHiAsync(string headParameter, string queryParameter, string optionalQuery)
+        public virtual async Task<Response<Thing>> SayHiAsync(string headParameter, string queryParameter, string optionalQuery = null)
```

Comment:
do we have an issue to track the item about adding cancellationToken?
Also maybe this optionality of RequestContext should also have some special mechanism in MGC - because MGC does not have cancellationToken, the optionality of parameters should be very simple to write for convenience methods but that simple logic might not apply to us.

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/47073

Reply:
This needs to be fixed in MGC - https://github.com/microsoft/typespec/issues/4991